src/modules/editor/sagas.ts

```diff
@@ -205,3 +213,20 @@ function* handleDropItem(action: DropItemAction) {
     yield put(addItem(asset, position))
   }
 }
+
+function* handleScreenshot(_: ScreenshotAction) {
+  try {
+    const screenshot = yield call(() => editorWindow.editor.takeScreenshot())
+    if (screenshot) {
+      const thumbnail = yield call(() => resizeScreenshot(screenshot, 246, 182))
+      if (thumbnail) {
+        const currentProject = yield select(getCurrentProject)
+        if (currentProject) {
+          yield put(editProject(currentProject.id, { thumbnail }))
+        }
+      }
+    }
+  } catch (e) {
```

Comment:
Since there's a `delay(500)` before dispatching the screenshot action, it could happen that the user leaves the editor page before the screenshot is taken and potentially throw, that's why I wrapped it in a try/catch

Reply:
gotcha!